python/lsst/daf/butler/core/storageInfo.py

```diff
@@ -0,0 +1,51 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from .utils import slotValuesAreEqual
+
+__all__ = ("StorageInfo")
+
+
+class StorageInfo:
+    """Represents storage information.
+    """
+    __eq__ = slotValuesAreEqual
+    __slots__ = ("_datastoreName", "_md5", "_size")
+
+    def __init__(self, datastoreName, md5=None, size=None):
+        assert isinstance(datastoreName, str)
+        self._datastoreName = datastoreName
+        assert md5 is None or isinstance(md5, str)
```

Comment:
Are we using md5 because it is ubiquitous? Should it be datastore specific or must it be defined by gen3 butler to be md5? blake2 is meant to be as fast as md5 but better (and is part of python3.6). Does it need to be md5 because @MichelleGower requires that from NCSA?

Reply:
We are using `md5` because that is what @TallJimbo put in the schema description. But it could/should probably be `checksum`.

Reply:
I have no preference as to what kind of checksum we use.  Probably best to make the name generic.